devicetypes/plaidsystems/spruce-controller.src/spruce-controller.groovy

```diff
@@ -14,736 +10,570 @@
+	sendCmds.push(zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, duration, [destEndpoint: 1]))
+	return sendCmds
 }
-def z11on() {        
-    return manual() + "st cmd 0x${device.deviceNetworkId} 12 6 1 {}"    
+
+//------------------end commands----------------------------------//
+
+//get times from settings and send to controller, then start schedule
+def startSchedule(){
+	def startRun = false
+	def runTime, totalTime=0
+	def scheduleTimes = []
+
+	for (i in 1..16){
+		def endpoint = i + 1        
+		if (settings."z${i}" && settings."z${i}Duration" != null){
+			runTime = Integer.parseInt(settings."z${i}Duration")
+			totalTime += runTime
+			startRun = true
+
+			scheduleTimes.push(zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, runTime, [destEndpoint: endpoint]))
+		}
+		else {
+			scheduleTimes.push(zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, 0, [destEndpoint: endpoint]))
+		}
+	}
+	if (!startRun || totalTime == 0) return noSchedule()
+
+	//start after scheduleTimes are sent
+	scheduleTimes.push(zigbee.command(ON_OFF_CLUSTER, 1, "", [destEndpoint: 1]))
+	sendEvent(name: "status", value: "Scheduled for ${totalTime}min(s)", descriptionText: "Start schedule ending in ${totalTime} mins")
+	return scheduleTimes
 }
-def z11off() {	
-    "st cmd 0x${device.deviceNetworkId} 12 6 0 {}"    
+
+//write switch time settings map
+def settingsMap(WriteTimes, attrType){
+
+	def runTime
+	def sendCmds = []
+	for (endpoint in 1..17) {
+		if (WriteTimes."${endpoint}"){
+			runTime = Integer.parseInt(WriteTimes."${endpoint}")
+
+			if (attrType == ON_TIME_ATTRIBUTE) sendCmds.push(zigbee.writeAttribute(ON_OFF_CLUSTER, ON_TIME_ATTRIBUTE, DataType.UINT16, runTime, [destEndpoint: endpoint]))
+			else sendCmds.push(zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, runTime, [destEndpoint: endpoint]))
+		}
+	}
+	return sendCmds
 }
-def z12on() {	
-    return manual() + "st cmd 0x${device.deviceNetworkId} 13 6 1 {}"    
+
+//send switch time
+def writeType(endpoint, cycle){
+	zigbee.writeAttribute(ON_OFF_CLUSTER, ON_TIME_ATTRIBUTE, DataType.UINT16, cycle, [destEndpoint: endpoint])
 }
-def z12off() {    
-    "st cmd 0x${device.deviceNetworkId} 13 6 0 {}"    
+
+//send switch off time
+def writeTime(endpoint, runTime){
+	zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, runTime, [destEndpoint: endpoint])
 }
-def z13on() {	
-	return manual() + "st cmd 0x${device.deviceNetworkId} 14 6 1 {}"   
+
+//set reporting and binding
+def configure() {
+	// Device-Watch allows 2 check-in misses from device, checks every 2 hours
+	sendEvent(name: "DeviceWatch-DeviceStatus", value: "online")
+	sendEvent(name: "healthStatus", value: "online")
+	sendEvent(name: "DeviceWatch-Enroll", value: 2* 60 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	configureHealthCheck()
```

Comment:
This is no longer necessary. 

just sending

`[name: "checkInterval", value: hcIntervalMinutes * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID]]`

and having a defined `ping` method should get you what you want. You can also send a similar event for all the child devices as long as they each have a ping method defined as well and then everything will be checked automatically if we haven't heard anything from the device itself.